src/Glide64/3dmath.cpp

```diff
@@ -190,16 +190,37 @@ void InverseTransformVectorC (float *src, float *dst, float mat[4][4])
 
 void MulMatricesC(float m1[4][4],float m2[4][4],float r[4][4])
 {
-  for (int i=0; i<4; i++)
-  {
-    for (int j=0; j<4; j++)
+    float row[4][4];
+    register unsigned int i, j;
+
+    for (i = 0; i < 4; i++)
+        for (j = 0; j < 4; j++)
+            row[i][j] = m2[i][j];
+    for (i = 0; i < 4; i++)
     {
-      r[i][j] = m1[i][0] * m2[0][j] +
-                m1[i][1] * m2[1][j] +
-                m1[i][2] * m2[2][j] +
-                m1[i][3] * m2[3][j];
+        float leftrow[4], destrow[4];
```

Comment:
I agree, there is a lot of SSE in many emulators. It's a shame as often, emulators whose code will remains for years have bad portability. It's also a reason why I tent to prefer shortness and brevity over _compiler compliant_ writing: Emulator code should focus on readability over performance as it make life of potential future contributors easier.

I'm not a compiler expert but if we convert this:
```c
for (j = 0; j < 4; j++)
    summand[0][j] = leftrow[0] * row[0][j];
for (j = 0; j < 4; j++)
    summand[1][j] = leftrow[1] * row[1][j];
for (j = 0; j < 4; j++)
    summand[2][j] = leftrow[2] * row[2][j];
for (j = 0; j < 4; j++)
    summand[3][j] = leftrow[3] * row[3][j];
```
To this:
```c
for (j = 0; j < 4; j++) {
    summand[0][j] = leftrow[0] * row[0][j];
    summand[1][j] = leftrow[1] * row[1][j];
    summand[2][j] = leftrow[2] * row[2][j];
    summand[3][j] = leftrow[3] * row[3][j];
}
```
Are we sure the compiler will not detect vectorizable code and provide the same result than the first case?

I'm just curious.